capstone/src/components/User/Adventurer/ExperienceInfo/ExperienceInfo.jsx

```diff
@@ -6,16 +6,42 @@ import {
   Image,
   VStack,
 } from '@chakra-ui/react';
-import { useState } from 'react';
+import axios from 'axios';
+import { useState, useContext, useEffect } from 'react';
+import UserContext from '../../../../Contexts/UserContext';
+import RateExperience from './RateExperience';
 
 const API_KEY = process.env.REACT_APP_GOOGLE_MAPS_API_KEY;
+const baseURL = process.env.REACT_APP_BASE_URL;
+
+async function getIsRated(username, experienceId) {
+  const values = { username, experienceId };
+  return axios.post(`${baseURL}/review/isRated`, values);
+}
+
+async function submitRate(username, experienceId, reviews) {
+  const values = { username, experienceId, reviews };
```

Comment:
Usually the username/userid is set on the session cookie or through a header. You shouldn't need to explicitly pass this in, since that would let you easily impersonate someone else.